tests/python_package_test/test_engine.py

```diff
@@ -1434,25 +1431,220 @@ def decreasing_metric(preds, train_data):
 
         def constant_metric(preds, train_data):
             return ('constant_metric', 0.0, False)
-
         # test that all metrics are checked (default behaviour)
+        params = {
+            'objective': 'regression',
+            'metric': 'None',
+            'verbose': -1
+        }
         gbm = lgb.train(params, lgb_train, num_boost_round=20, valid_sets=[lgb_eval],
                         feval=lambda preds, train_data: [decreasing_metric(preds, train_data),
                                                          constant_metric(preds, train_data)],
                         early_stopping_rounds=5, verbose_eval=False)
         self.assertEqual(gbm.best_iteration, 1)
-
         # test that only the first metric is checked
+        try:
```

Comment:
Replace this with https://docs.scipy.org/doc/numpy/reference/generated/numpy.testing.assert_raises_regex.html#numpy.testing.assert_raises_regex